kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/CreateQuizSection.vue

```diff
@@ -503,6 +511,13 @@
         ];
       },
     },
+    created() {
+      const { query } = this.$route;
+      if (query.snackbar) {
+        delete query.snackbar;
+        this.$router.replace({ query: { snackbar: null } });
+      }
```

Comment:
I don't particularly like this solution but I had a hard time working out a way to show the snackbars during side-panel closing. I tried setTimout-ing it so that the snackbar could show before the panel closed but it was a weird pause between saying "close this" and seeing it closed. 

In any case, I'd appreciate thoughts on how this might be better approached.

Reply:
I might need more detail on the desired behaviour here to make a recommendation - is it that you are using the snackbar for confirmation before closing?

Reply:
The snackbar message like "Changes saved successfully" gets lost and doesn't show. So like

- In a side panel, click "Save"
- We call to show the snackbar at the end of the saving function
- The saving function also then calls to close the panel
- Closing the panel (or changing the route?) causes the snackbar not to show

So the solution here using the query params to pass the snackbar message around in this case feels kind of wack.

Like maybe I could try something like... provide a generic "showSnackbar" method from the quiz page root that we inject in child components so that maybe calling the function in the scope of the parent would allow us to show the snackbar while or after closing the side panel without having to set the query param.

Not sure if that would be more/less readable than this but I feel like this might be a bit brittle anyhow.

Reply:
So - I imagine this is happening because part of our coach global router hooks is to clear the snackbar?

https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/coach/assets/src/app.js#L52

So, maybe it would be simpler to add an exception here for this case?

Reply:
This might be a place where your use of nested routes might be an advantage - if the parent route hasn't changed, then don't trigger the clear.

Reply:
Omg the global router strikes again 😭 - thanks @rtibbles